### PR TITLE
feat(gateway): Authority Spine Phase 1 — scope attenuation, revocation, and capability truth

### PR DESCRIPTION
**Pull request by fahertym**
## What

Authority Spine Phase 1. Introduces `SessionAuthority` — one explicit, test-constructible composition boundary owning gateway session authority — and lands #2436 and #2437 through it rather than as isolated patches.

## Why

Both issues are instances of one structural failure: a capability exists in a crate, is unit-tested against a miniature composition, and is not installed in the assembled runtime, while docs describe library capability as a runtime guarantee. RPC revocation machinery was constructed as `None`; `token_expiry_hours` was parsed but every gateway credential lived one hour.

## Invariants enforced by this branch

- **Attenuation:** `issued ⊆ issuer ∩ flow_allowed ∩ requested`; every term is a ceiling, never a grant.
- **Expiration:** the configured lifetime bounds both issuance and acceptance **on the gateway surface** — `SessionAuthority::verify` refuses a credential whose `exp − iat` exceeds the configured lifetime and, independently, one whose expiry lies more than one lifetime from now (a forward-dated `iat` buys nothing); client-visible lifetime fields report the installed authority lifetime; JWT expiry has no hidden leeway. This is **not** a universal cross-surface lifetime bound — the RPC surface shares the signing secret and is not bounded; see "What this does not do".
- **Revocation:** a credential can be individually withdrawn and is revalidated on HTTP requests, RPC requests, and every protected WebSocket event/backfill operation.
- **Truth:** capability reports derive from the constructed authority; required guarantees fail assembly; gateway actor-active state is set only after initialization and bind acknowledgement and is cleared on termination.

## #2436 — QR session-approve privilege escalation

`POST /v1/sessions/{id}/approve` previously checked only matching `coop_id`, then minted six broad scopes. The flow ceiling is now intersected with the approver's own scopes through `SessionAuthority::issue_delegated`. A holder with no delegable scope receives 403. The unmounted duplicate approval handler was removed.

The route remains attenuated rather than deleted. Requiring a dedicated approval capability is an institutional policy decision outside this PR.

## #2437 — revocation and lifetime

- Every gateway mint path bottoms out in `issue_entity_token`, which now emits a random `jti`.
- HTTP bearer middleware, in-band WebSocket auth, and hand-authenticated SDIS paths verify through `SessionAuthority` and fail closed when it is absent or revocation state is unreadable.
- The daemon installs RPC revocation with the same durable store used by the gateway.
- Gateway and RPC positive caches fall through to the shared store on misses, so cross-surface revocation is visible without restart; errors deny.
- The canonical unconfigured/default session lifetime is 24 hours across `AuthManager`, `TokenLifetimePolicy`, embedded gateways, and daemon `GatewayConfig`. Authority assembly rejects issuer/policy lifetime divergence.
- `/auth/verify`, invite join, and QR-session responses derive lifetime from the installed authority.
- WebSockets retain the credential and revalidate after subscription setup and before each protected event and backfill event. Revocation or expiry closes the socket before protected delivery.

## Operator-visible changes

1. **BREAKING:** a daemon with the durable revocation store runs the institutional profile and rejects pre-`jti` credentials immediately; holders must re-authenticate.
2. Default session lifetime changes from the formerly effective 1 hour to the configured/canonical 24 hours. Set `token_expiry_hours = 1` to retain one-hour sessions.
3. New boot dependency: `<store_path>/auth-revocation`. Failure to open or read required revocation state is fail-closed.
4. A gateway startup failure does not abort the daemon, but the supervisor now waits for readiness and reports the gateway actor inactive rather than claiming it started.
5. Verification of a non-revoked credential performs a point read from the shared revocation store. No negative cache is used because it would create a stale-revocation window.
6. A credential whose lifetime exceeds the deployment's configured session lifetime is refused at acceptance with a message naming the bound (previously the signed expiration was honored). This applies to trusted-local mints and to credentials issued before an operator lowered `token_expiry_hours`. `icnctl auth token --local-mint` takes `--expiry-hours` — validated through the same `TokenLifetimePolicy` as the gateway's own configuration — for deployments configured shorter than the canonical 24-hour default.

## Validation

- `cargo fmt --all --check`
- `cargo clippy --workspace --all-targets -- -D warnings`
- Full `icn-gateway --features sled-storage`: 712 unit tests plus all integration suites, including three real-TCP production-handler WebSocket revocation/expiry tests.
- Full `icn-core`, `icn-rpc`, `icnctl`, and `icn-obs` package suites.
- NYCN `nycn_bootstrap_apply_round_trip` executes and passes through the corrected router.
- `session_authority_enforcement`: 17 middleware/composition cases, including live gateway↔RPC revocation in both directions and exact expiry boundary.
- OpenAPI drift comparison, route-inventory check, agent-context-spine check, doc control, compliance, and readiness-overclaim checks pass.
- Repository freshness currently reports unchanged sections that crossed their age threshold; current `main` reproduces that pre-existing documentation debt. It is not papered over here.

## What this does not do

- Does not extract the full production router; #2421 remains the production-assembly closure.
- Does not add a gateway HTTP revocation route; revocation is reached through RPC `auth.revoke`.
- Does not decide who may approve or revoke, what evidence revocation leaves, or member recourse.
- Attenuation is enforced for delegated session approval; invite, SDIS enrollment, and trusted-local mint remain flow-chosen and capability reporting stays `Degraded` rather than overclaiming.
- Does not introduce a speculative revocation-store performance redesign. The correctness-preserving point read remains until evidence justifies a coherent invalidation mechanism.
- **Does not make the lifetime bound universal across surfaces.** The daemon derives the RPC signing key from the same `gateway.jwt_secret` (`supervisor/init_rpc.rs`), `RpcTokenClaims` does not reject unknown fields, and every field it requires is present in a gateway `TokenClaims` — so a gateway credential is structurally verifiable on the RPC surface. `icn-rpc`'s `verify_token` checks signature, `exp`, and revocation but applies no configured-lifetime bound, and its own issuance lifetime is a hardcoded 24 hours. Proven directly on this branch: against a one-hour deployment, a validly signed 24-hour co-issued credential is refused by `SessionAuthority::verify` and accepted by `RpcAuthManager::verify_token`. This is **pre-existing on `main`** — before this branch neither surface applied an acceptance bound — and is not a regression introduced here, but it means lowering `token_expiry_hours` shortens gateway acceptance without shortening an already-issued credential on RPC. Shared revocation, which *did* land here, remains the instrument that invalidates a credential on both surfaces. Closing it requires bounding RPC acceptance **and** issuance together (bounding acceptance alone would refuse the RPC manager's own freshly minted tokens) with an explicit credential-invalidation migration. Tracked in #2445.

## Review disposition

Commit `7cf8c478` addresses startup truth, client-visible TTL truth, WebSocket continuing authorization, bootstrap-router composition, split defaults, and live cross-surface RPC/gateway revocation. The JTI path retains `rand::rng()` because pinned rand 0.9 contractually provides a ChaCha12 `CryptoRng` seeded/reseeded from `OsRng`; a compile-time `CryptoRng` bound and code documentation now make that guarantee explicit. The non-revoked store read is retained deliberately for correctness.

Commit `9287b79e` closes the remaining P2 (local-mint lifetime): the configured lifetime is now an acceptance bound enforced at `SessionAuthority::verify` — both the `exp − iat` check and the remaining-window check — and the trusted-local mint gained `--expiry-hours` (validated through the shared `TokenLifetimePolicy`) so it stays usable on shorter-lifetime deployments. Regression matrix: co-issued 24-hour credential vs a 1-hour deployment → 401 through the real `jwt_auth` middleware, bounded re-mint → 200; exact-equality boundary; forward-dated `iat`; icnctl knob applies/defaults/rejects-invalid. The `Compare Against Base` failure is disposed as measurement noise with a code-path non-overlap analysis in the PR conversation; the revocation point read is not weakened.

Commit `a846a3dd` is a claims-boundary correction with no behavior change. An independent adversarial closure review found the Expiration invariant was stated as a universal cross-surface guarantee when it holds on the gateway surface only; the architecture note and this description now say so, and the RPC gap is tracked in #2445. The gateway invariant itself was re-verified rather than assumed: attenuation is a true double intersection that errors on the empty set; the acceptance bound was probed at `exp = u64::MAX`, `iat = 0`, `iat > exp`, `iat = u64::MAX`, exact equality, and one second over — no `iat` manipulation can yield an accepted window longer than the configured lifetime, and malformed orderings fail short rather than long; the only production JWT verification in the gateway crate is inside `SessionAuthority::verify`, so HTTP, WebSocket revalidation, and the hand-authenticated SDIS paths all inherit the bound.

Refs #2436 #2437 #2421 #2445

### CODE REVIEW

File: icn/crates/icn-core/src/supervisor/init_gateway.rs [hunk 1]
```diff
@@ -273,6 +284,34 @@ pub fn spawn_gateway(config: &GatewayConfig, data_dir: PathBuf, handles: Gateway
...
+                }
+                Err(e) => {
+                    // Misconfigured lifetime is an operator error: refuse to
+                    // start rather than silently fall back to a different
+                    // lifetime than the one configured.
+                    warn!("Gateway not started: invalid token_expiry_hours — {}", e);
+                    icn_obs::metrics::supervisor::error_inc("gateway_server");
+                    return;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-core/src/supervisor/init_gateway.rs]: Fixed in 7cf8c478. token_expiry_hours is validated before thread creation, and a two-way readiness handshake delays actor_spawned/actor_active until authority assembly, storage initialization, and bind succeed; actor_active is cleared when the server terminates. Regression: supervisor::init_gateway::tests::invalid_token_lifetime_is_rejected_before_gateway_spawn plus server::tests::startup_signal_reports_failure_before_readiness. Full icn-core suite passes.

File: icn/crates/icn-gateway/src/session_authority.rs [hunk 2]
```diff
@@ -0,0 +1,904 @@
...
+        })?;
+
+        if present.is_some() {
+            if let Ok(mut cache) = self.cache.write() {
+                cache.insert(jti.to_string());
+            }
+            return Ok(true);
+        }
+        Ok(false)
+    }
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-gateway/src/session_authority.rs:302]: Disposition in 7cf8c478: retained deliberately for correctness. A positive-cache miss is not evidence of validity because RPC and gateway share the store; a negative cache would create a stale-revocation acceptance window. Both verifiers now perform the same fail-closed point read, and AUTHORITY_SPINE.md documents the cost and rationale. The live cross-surface test constructs RPC before the gateway revoke and proves immediate rejection without restart. No demonstrated bottleneck justifies weakening the guarantee.

File: icn/crates/icn-gateway/src/auth.rs [hunk 3]
```diff
@@ -400,6 +436,18 @@ impl AuthManager {
...
+    ///
+    /// Randomness — not a counter or a hash of the claims — so two credentials
+    /// minted for the same subject in the same second are still independently
+    /// revocable, and so an id reveals nothing about issuance order or volume.
+    fn generate_jti() -> String {
+        use rand::Rng;
+        let mut rng = rand::rng();
+        let bytes: [u8; 16] = rng.random();
+        hex::encode(bytes)
+    }
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-gateway/src/auth.rs:450]: Addressed in 7cf8c478 without substituting meaningless randomness tests. Pinned rand 0.9 documents rand::rng() as ChaCha12, automatically seeded and periodically reseeded from OsRng. The helper now states that contract and applies a compile-time CryptoRng bound, so a non-cryptographic replacement cannot silently compile for JTI or challenge generation. Full gateway tests and workspace clippy pass.

File: icn/crates/icn-gateway/src/session_authority.rs [hunk 4]
```diff
@@ -0,0 +1,904 @@
...
+
+/// Upper bound on any configured session lifetime.
+///
+/// A credential that cannot be revoked for longer than this is an unbounded
+/// liability; 30 days is the outer edge at which an operator should be issuing
+/// a new credential rather than extending an old one.
+pub const MAX_TOKEN_TTL: Duration = Duration::from_secs(30 * 24 * 3600);
+
+/// Fallback lifetime when a deployment supplies no configuration.
+pub const DEFAULT_TOKEN_TTL: Duration = Duration::from_secs(3600);
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-gateway/src/session_authority.rs]: Fixed in 7cf8c478. DEFAULT_TOKEN_TTL is now the canonical 24-hour default used by AuthManager, TokenLifetimePolicy, embedded GatewayServer paths, and GatewayConfig. SessionAuthority::new rejects an issuer/policy TTL mismatch. Regressions cover the shared unconfigured default, explicit non-default policy, and mismatch refusal.

File: icn/crates/icn-gateway/src/server.rs [hunk 5]
```diff
@@ -697,9 +750,48 @@ impl GatewayServer {
...
+        // constructible, inspectable, and testable through the same path
+        // production uses. `SessionAuthority::new` runs the profile's startup
+        // invariants and refuses to assemble a deployment whose required
+        // guarantees are missing.
+        let lifetime = self.token_lifetime.unwrap_or_default();
         let auth_manager = Arc::new(
-            AuthManager::new(self.jwt_secret).with_self_asserted_coop(allow_self_asserted_coop),
+            AuthManager::new(self.jwt_secret)
+                .with_self_asserted_coop(allow_self_asserted_coop)
+                .with_token_ttl(lifetime.ttl()),
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-gateway/src/server.rs:788]: Fixed in 7cf8c478. /auth/verify now resolves SessionAuthority and reports authority.lifetime(); the non-default 2-hour regression decodes the returned credential and asserts exp-iat equals expires_in. The same correction was applied to invite join, while QR session approval already reports the authority lifetime. JWT verification also disables the library default expiry leeway so reported exp is the actual acceptance boundary.

File: icn/crates/icn-gateway/src/websocket.rs [hunk 6]
```diff
@@ -111,7 +111,11 @@ impl WsSession {
 
     /// Authenticate user with JWT token
     fn authenticate(&mut self, token: &str, ctx: &mut <Self as Actor>::Context) {
-        match self.auth_manager.verify_token(token) {
+        // Verify through the session authority, NOT the bare AuthManager: this
+        // surface is mounted outside `jwt_auth` and authenticates in-band, so
+        // without this it would honor revoked and jti-less credentials that
+        // every wrapped HTTP route rejects (issue #2437).
+        match self.authority.verify(token) {
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-gateway/src/websocket.rs:123]: Fixed in 7cf8c478 with operation-level revalidation rather than a timer window. WsSession retains the credential, revalidates after async subscription setup, before every live event, before backfill, before every backfill event, and before completion; revocation/expiry clears the receiver and stops the actor before protected delivery. websocket_authority_enforcement.rs drives a real TCP server through the production handler/actor and proves no live event after revoke, no backfill after revoke, and no live event after expiry.

File: icn/crates/icn-gateway/src/middleware.rs [hunk 7]
```diff
@@ -10,20 +10,32 @@ use std::future::{ready, Ready};
...
-        Some(mgr) => mgr.clone(),
+    // Verify through the session authority — the composition boundary that owns
+    // signature, expiry AND revocation (issue #2437). Resolving it here, at the
+    // single middleware every authenticated route is wrapped with, is what makes
+    // revocation unavoidable: a route cannot opt out by forgetting to check.
+    //
+    // Fail closed when it is absent. Falling back to a bare `AuthManager` would
+    // silently restore the pre-#2437 behavior (valid signature ⇒ authorized) in
+    // exactly the misassembly this work exists to prevent.
+    let authority = match req.app_data::<actix_web::web::Data<Arc<SessionAuthority>>>() {
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-gateway/src/middleware.rs:31]: Fixed in 7cf8c478. The icnctl bootstrap router now installs SessionAuthority, and auth/invite/enrollment handlers consume that same authority rather than separate AuthManager app_data, reducing the composition to one required object. nycn_bootstrap_apply_round_trip executes successfully, and the full icnctl suite passes.

File: icn/crates/icn-gateway/src/auth.rs [hunk 8]
```diff
@@ -185,7 +202,7 @@ impl AuthManager {
             challenges: Arc::new(RwLock::new(HashMap::new())),
             jwt_secret,
             challenge_ttl: Duration::from_secs(300), // 5 minutes
-            token_ttl: Duration::from_secs(3600),    // 1 hour
+            token_ttl: crate::session_authority::DEFAULT_TOKEN_TTL,
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-gateway/src/auth.rs:205]: Fixed in `9287b79e` — in both directions, not only by re-plumbing the flag.

**Acceptance (structural).** `SessionAuthority::verify` now enforces the configured lifetime as an *acceptance* bound at the one chokepoint every surface already routes through (HTTP middleware, WebSocket revalidation, hand-authenticated SDIS paths):

1. a credential whose `exp − iat` exceeds the configured lifetime is refused loudly, naming the bound — not clamped, which would silently shorten a session its holder was told was longer;
2. independently, a credential whose `exp` lies more than one configured lifetime from now is refused, so a fabricated or forward-dated `iat` cannot buy extra validity.

This makes "the configured lifetime bounds issuance **and acceptance**" true by construction for any co-issuer holding the signing secret — today's `icnctl --local-mint`, and any future drift in either direction (including previously-issued long credentials after an operator lowers `token_expiry_hours`).

**Issuance (usability).** `icnctl auth token --local-mint` now takes `--expiry-hours`, validated through the same `TokenLifetimePolicy::from_hours` the gateway applies to its own configuration, so the two surfaces cannot disagree about what a legal lifetime is. The default remains the canonical 24 hours — identical behavior on unconfigured deployments. On your 1-hour example the operator mints `--expiry-hours 1`; if they forget, the gateway's rejection names the configured bound.

**Regression coverage.** `co_issued_credential_exceeding_the_configured_lifetime_is_rejected` drives exactly your scenario (24h local-mint-shaped credential vs. a 1-hour deployment) through the real `jwt_auth` middleware → 401, then the bounded re-mint → 200. Unit tests pin the exact-equality boundary, the forward-dated case, and the icnctl knob (applies / defaults-to-canonical / rejects zero-and-excessive through the shared validation). The existing expiry-boundary test now also pins the equality case through the middleware.

Trusted-local mint remains a distinct authority class in *provenance* (it exists only for an operator who already holds the gateway's signing secret), but deliberately **not** in lifetime: deployment policy is a ceiling on what is accepted, and an operator who wants longer sessions changes `token_expiry_hours` — the configuration is already theirs.

**Comment by fahertym** [comment]
**Benchmark disposition (`Compare Against Base`, non-required).**

The 13 reported regressions are all in benches whose code paths this PR does not touch: postcard/message/ledger serialization, `ledger_crypto/verify` (Ed25519), gossip compression, and anchor/charter lookups. The diff modifies gateway auth/session composition, supervisor wiring, RPC auth, icnctl, and one metrics module — none of the regressed benches exercise those crates, and no bench in `BENCH_TARGETS` exercises gateway credential verification at all, which is where this PR's one deliberate cost lives (a revocation-store point read per verification, documented in the PR body and retained for correctness — a negative cache would create a stale-revocation window).

Conclusion: shared-runner measurement noise, the same shape previously observed on this workflow against unrelated serialization benches. The push addressing the local-mint lifetime review re-runs the comparison. The revocation read will not be weakened to green a non-required check; if a future *auth-path* benchmark shows a real cost, that is the honest place to measure it.

**Comment by fahertym** [comment]
## Adversarial closure review — finding and disposition

An independent review pass re-derived this PR's invariants rather than trusting the prior round. One truth-boundary defect surfaced, in the description rather than the implementation.

**The gateway lifetime invariant holds.** Re-verified, not assumed:

- attenuation is a genuine double intersection (`issuer ∩ flow`, then `∩ requested`) that errors on the empty set — no union, no wildcard expansion, no default grant;
- the acceptance bound was probed at `exp = u64::MAX`, `iat = 0`, `iat > exp`, `iat = u64::MAX`, exact equality, and one second over the bound. No `iat` manipulation yields an accepted window longer than the configured lifetime; malformed orderings fail *short*, never long, because the remaining-window check bounds `exp` independently of what `iat` claims;
- the only production JWT verification in `icn-gateway` is inside `SessionAuthority::verify`, so HTTP bearer routes, WebSocket revalidation, and the hand-authenticated SDIS paths all inherit the bound; `jwt_auth` fails closed when the authority is absent.

**Universal cross-surface lifetime enforcement does not hold, and this PR said it did.** The daemon derives the RPC signing key from the same `gateway.jwt_secret` (`supervisor/init_rpc.rs`), `RpcTokenClaims` does not reject unknown fields, and every field it requires is present in a gateway `TokenClaims`. `icn-rpc`'s `verify_token` checks signature, `exp`, and revocation — no configured-lifetime bound — and its own issuance lifetime is a hardcoded 24 hours that never reads `token_expiry_hours`.

Proven directly on this branch, one-hour deployment vs a validly signed 24-hour co-issued credential:

| Surface | Verdict |
|---|---|
| `SessionAuthority::verify` | refused — `credential lifetime exceeds this deployment's configured session lifetime (3600 seconds)` |
| `RpcAuthManager::verify_token` | **accepted** — `exp − iat = 86400` |

This corrects my earlier reply on the local-mint thread, which said the bound was true "for any co-issuer holding the signing secret." That is accurate for the gateway surface and overstated for the system: RPC does not route through `SessionAuthority::verify`.

**This PR does not regress RPC.** Before this branch neither surface applied an acceptance bound; the branch strictly improves the gateway and leaves RPC as it found it. The operational consequence worth knowing: lowering `token_expiry_hours` shortens gateway acceptance immediately but does not shorten an already-issued credential on RPC. Shared revocation — which *did* land here — remains the instrument that reaches both surfaces.

**Disposition.** Not fixed here, deliberately. Bounding RPC acceptance alone would make the RPC manager reject its own freshly minted 24-hour tokens; a correct fix changes issuance and acceptance together and carries a credential-invalidation migration. That is #2445. Commit `a846a3dd` is a claims-boundary correction only — no behavior change — scoping the Expiration invariant to the gateway surface in both `docs/architecture/AUTHORITY_SPINE.md` and this description, and naming the RPC gap with its evidence.

The benchmark comparison remains non-required and non-overlapping (regressions confined to serialization, ledger-crypto, gossip-compression, and naming-lookup benches; no auth benchmark exists in `BENCH_TARGETS`). The revocation point read is unchanged.

File: icn/crates/icn-rpc/src/auth.rs [hunk 9]
```diff
@@ -196,30 +196,44 @@ impl<S: Store> TokenRevocationList<S> {
...
             .cache
             .read()
             .map(|cache| cache.contains(jti))
-            .unwrap_or_else(|e| {
-                tracing::warn!(error = %e, jti = %jti, "Cache lock poisoned in is_revoked, failing open");
-                false
-            });
+            .map_err(|e| {
+                AuthError::InternalError(format!("Revocation cache lock poisoned: {e}"))
+            })?;
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-rpc/src/auth.rs:219]: Valid — fixed in `39f5fbf4`.

`icn_rpc_revocation_errors_total` is described as `"Total number of revocation operation errors (storage, serialization, lock)"` (`icn-obs/src/metrics/rpc.rs`), and `revoke()` in this same file already increments it on a poisoned cache write (`auth.rs`, the `Lock poisoned` arm). `is_revoked` was the outlier: both of its lock-poisoned arms — the cache read and the cache write that caches a store hit — returned the error without counting it. Since this PR is what rewrote `is_revoked` from fail-open `-> bool` to fail-closed `-> Result<bool, AuthError>`, those two arms are lines this branch introduced, so the gap is in scope here rather than pre-existing.

Both now increment before returning. Authorization behavior is unchanged — a poisoned lock still denies — but the denial is now visible to the metric that claims to cover it. That matters for exactly the diagnosis case you name: a fail-closed denial that never reaches a counter is invisible while an operator is trying to explain a wave of 401s.

`load_cache()` has the same shape on its scan and lock arms, but it is pre-existing code this branch does not touch, so it is left alone rather than fixed drive-by.

No regression test accompanies this: asserting a counter increment needs a metrics recorder this workspace has no test harness for, and poisoning the lock requires a panic across a thread boundary — building that machinery inside a security PR would cost more than the two-line correctness it guards. The behavior these arms already had (fail closed on a poisoned lock) is covered by the existing suite. Flagging that honestly rather than implying coverage that does not exist.

File: icn/crates/icn-rpc/src/auth.rs [hunk 10]
```diff
@@ -196,30 +196,44 @@ impl<S: Store> TokenRevocationList<S> {
...
-        is_revoked
+        let present = self.store.get(&Self::make_key(jti)).map_err(|e| {
+            counter!("icn_rpc_revocation_errors_total").increment(1);
+            AuthError::InternalError(format!("Failed to read token revocation state: {e}"))
+        })?;
+
+        if present.is_some() {
+            let mut cache = self.cache.write().map_err(|e| {
+                AuthError::InternalError(format!("Revocation cache lock poisoned: {e}"))
+            })?;
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-rpc/src/auth.rs:235]: Valid — fixed in `39f5fbf4`.

`icn_rpc_revocation_errors_total` is described as `"Total number of revocation operation errors (storage, serialization, lock)"` (`icn-obs/src/metrics/rpc.rs`), and `revoke()` in this same file already increments it on a poisoned cache write (`auth.rs`, the `Lock poisoned` arm). `is_revoked` was the outlier: both of its lock-poisoned arms — the cache read and the cache write that caches a store hit — returned the error without counting it. Since this PR is what rewrote `is_revoked` from fail-open `-> bool` to fail-closed `-> Result<bool, AuthError>`, those two arms are lines this branch introduced, so the gap is in scope here rather than pre-existing.

Both now increment before returning. Authorization behavior is unchanged — a poisoned lock still denies — but the denial is now visible to the metric that claims to cover it. That matters for exactly the diagnosis case you name: a fail-closed denial that never reaches a counter is invisible while an operator is trying to explain a wave of 401s.

`load_cache()` has the same shape on its scan and lock arms, but it is pre-existing code this branch does not touch, so it is left alone rather than fixed drive-by.

No regression test accompanies this: asserting a counter increment needs a metrics recorder this workspace has no test harness for, and poisoning the lock requires a panic across a thread boundary — building that machinery inside a security PR would cost more than the two-line correctness it guards. The behavior these arms already had (fail closed on a poisoned lock) is covered by the existing suite. Flagging that honestly rather than implying coverage that does not exist.

File: docs/DOCUMENT_REGISTRY.md [hunk 11]
```diff
@@ -1,7 +1,7 @@
 ---
 Status: descriptive
 Canonical: no
-Last Reviewed: 2026-07-18
+Last Reviewed: 2026-07-19
 ---
```
[the remark this answers is not in the document]
Comment (fahertym) [reply docs/DOCUMENT_REGISTRY.md:5]: Valid — fixed in `9a1d8f29`.

Confirmed: the header read `2026-07-19` while this branch had already moved the `AUTHORITY_SPINE.md` registry row to `2026-07-21`, so the generated companion described a snapshot older than the registry it summarizes.

Regenerated through the documented path rather than hand-edited, since this file is a generated artifact:

```
python3 docs/scripts/doc_control_check.py --repo . --registry docs/registry.toml \
  --write-document-registry docs/DOCUMENT_REGISTRY.md
```

The generator stamps the header itself, and it produced exactly the `2026-07-21` you expected. Corpus counts are unchanged at 953 — only a row's dates moved on 07-21, not the corpus, so the counts this PR introduced (952 → 953) were already correct.

One addition beyond what you flagged: `registry.toml`'s own row for `DOCUMENT_REGISTRY.md` is bumped in the same commit. Updating the header alone would have left the `yaml-mismatch` warning `doc_control_check` emits when a file's `Last Reviewed` disagrees with its registry row — the mirror image of the defect that blocked #2435. With both moved, enforcement warnings drop 68 → 67 and the `DOCUMENT_REGISTRY` mismatch is cleared rather than relocated.

File: docs/architecture/AUTHORITY_SPINE.md [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+**Which deployments actually get which profile today**, since a profile
+document that does not say this is exactly the kind of unbacked capability claim
+this note argues against: the profile is *inferred from whether a revocation
+store was supplied*, not declared by operator configuration. The daemon always
+supplies one, so **every daemon deployment — including the portable evaluator
+appliance — runs `Institutional`**. `PortableEvaluator` is reached only by
+embedded/test callers that construct a `GatewayServer` without a store. Two
+consequences worth naming: the evaluator appliance does get durable revocation
+(it has a real store), and a deployment whose keystore fails to unlock takes a
+path that supplies no store, silently *downgrading* the authority guarantee.
```
[the remark this answers is not in the document]
Comment (fahertym) [reply docs/architecture/AUTHORITY_SPINE.md]: Valid — fixed in `c5409697`.

You're right, and the paragraph refuted itself in its own last sentence. Checked against the code rather than just making it internally consistent: `revocation_store` is an `Option<Arc<dyn Store>>` on the gateway handles, assigned in `supervisor/lifecycle.rs` once the component set completes, and consumed in `init_gateway.rs` with `if let Some(..)`. So the supply is conditional, and `Institutional` is a property of a daemon that got far enough — not of being a daemon.

Rewritten to say that directly: a fully started daemon (including the evaluator appliance) opens `<store_path>/auth-revocation` and runs `Institutional` with durable revocation; any path that misses that assignment yields `PortableEvaluator` — embedded and test callers, **and equally a daemon run that never reaches it**, such as one whose keystore does not unlock.

I also kept the consequence you were circling explicit rather than leaving it implied: because the profile is inferred and never declared, a daemon can downgrade its own authority guarantee without an operator asking for it, and the only thing that says so is the capability report. The operator-declared-profile follow-up now states what it would actually buy — failing instead of quietly becoming disposable.

File: icn/crates/icn-gateway/tests/session_authority_enforcement.rs [hunk 13]
```diff
@@ -0,0 +1,760 @@
...
+                    web::get().to(|| async { HttpResponse::Ok().body("ok") }),
+                )
+                .wrap(auth_mw),
+        ),
+    )
+    .await;
+
+    let token = AuthManager::new(SECRET.to_vec())
+        .issue_token(&did(), "test-coop", scopes(&["coop:read"]))
+        .expect("mint");
```
[the remark this answers is not in the document]
Comment (fahertym) [reply icn/crates/icn-gateway/tests/session_authority_enforcement.rs:466]: Good catch — adopted in `bcd29220`.

You're right about the sensitivity gap, and it's worth stating precisely because it's the difference between a test that guards the invariant and one that only appears to. With nothing else in `app_data`, the fixture proved the middleware fails when *nothing* can verify a credential. The regression that matters is the opposite: a refactor adding an `.or_else(..)` fallback to a bare issuer would restore "valid signature ⇒ authorized" while every route still looked wrapped — and against the old fixture that refactor would still find no issuer, still fail closed, and the test would keep passing straight through the regression it exists to catch.

A usable `AuthManager` is now registered while `SessionAuthority` stays absent, in both shapes a reintroduction would plausibly take (`Data<AuthManager>` and `Data<Arc<AuthManager>>`), since a fallback could reach for either. The request must still be refused.

Verified the current behavior is correct rather than assuming it: `jwt_auth` resolves only `web::Data<Arc<SessionAuthority>>` and returns the "refusing to authenticate without revocation enforcement" error when it is missing — there is no fallback path today. Production registers no `AuthManager` in `app_data` at all, so these entries are deliberately inert; the test asserts they stay that way.

Test-only, no runtime change; the suite is unchanged at 17 passing.
